objects/src/accounts/account_id.rs

```diff
@@ -27,7 +28,10 @@ pub enum AccountType {
 ///  - 1 - only the account hash is stored on-chain which serves as a commitment to the account state.
 /// As such the three most significant bits fully describes the type of the account.
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
-pub struct AccountId(Felt);
+pub struct AccountId {
+    id: Felt,
+    seed: Option<Word>,
+}
```

Comment:
I agree with the sentiment here.  I'm not a huge fan of this solution.  We could consider moving it to a higher level - for example adding a new field (`account_seed: Option<Word>`) to the [`PreparedTransaction`](https://github.com/0xPolygonMiden/miden-base/blob/frisitano-create-account/objects/src/transaction/prepared_tx.rs#L15) object?

Reply:
I've updated [`PreparedTransaction`](https://github.com/0xPolygonMiden/miden-base/blob/frisitano-create-account/objects/src/transaction/prepared_tx.rs#L18) to include a field for `account_seed`.  I think this is fine for now.  However, in the future I think we should replace this with a more generic `TransactionExtriniscs` struct.  This can include fields which are relevant in certain circumstances - e.g. account creation / code updates. 

```rust
pub struct TransactionExtrinsics {
     account_seed: Option<Word>,
     updated_code: Option<ModuleAst>
     ...
}
```